worlds/smz3/__init__.py

```diff
@@ -377,6 +378,24 @@ def get_filler_item_name(self) -> str:
     def write_spoiler(self, spoiler_handle: TextIO):
             self.world.spoiler.unreachables.update(self.unreachable)
 
+    def JunkFillGT(self):
+        GTLocationNames = [loc.Name for loc in self.smz3World.GetRegion("Ganon's Tower").Locations]
+        for loc in self.locations.values():
+            if loc.name in GTLocationNames and loc.item is None:
+                poolLength = len(self.world.itempool)
+                # start looking at a random starting index and loop at start if no match found
+                for i in range(self.world.random.randint(0, poolLength), poolLength):
+                    if not self.world.itempool[i].advancement:
+                        itemFromPool = self.world.itempool.pop(i)
+                        break
+                else:
+                    for i in range(0, poolLength):
+                        if not self.world.itempool[i].advancement:
+                            itemFromPool = self.world.itempool.pop(i)
+                            break
+                self.world.push_item(loc, itemFromPool, False)
+                loc.event = itemFromPool.advancement
```

Comment:
itemFromPool.advancement should never be True.

However maybe adding the locations to excluded location already achieves what you want, and we don't need to do this manual fill anyway?

Reply:
I cant use excluded locations as I wouldnt be able to place keys in GT in pre_fill. Excluded locations add their rules before that. Also, its more in sync with the way its done in the original. And I might need that way of doing it when updating.